index.js

```diff
@@ -214,22 +272,56 @@ NYC.prototype.report = function (cb, _collector, _reporter) {
 
 NYC.prototype._loadReports = function () {
   var _this = this
-  var files = fs.readdirSync(this.tmpDirectory())
+  var files = fs.readdirSync(this.tempDirectory())
+
+  var cacheDir = _this.cacheDirectory()
 
-  return _.map(files, function (f) {
+  var loadedMaps = this.loadedMaps || (this.loadedMaps = {})
+
+  return files.map(function (f) {
+    var report
     try {
-      return JSON.parse(fs.readFileSync(
-        path.resolve(_this.tmpDirectory(), './', f),
+      report = JSON.parse(fs.readFileSync(
+        path.resolve(_this.tempDirectory(), './', f),
         'utf-8'
       ))
     } catch (e) { // handle corrupt JSON output.
       return {}
     }
+
+    if (!_this.enableCache) {
+      return report
+    }
+
+    Object.keys(report).forEach(function (relFile) {
```

Comment:
This logic is fairly complicated, I'd be tempted to pull it into a logically named "private" method (prefix with `_`).